pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -113,11 +120,56 @@ func (r *Cluster) ValidateUpdate(old runtime.Object) error {
 	return nil
 }
 
+func validateEksaVersionSkew(new, oldCluster *Cluster) field.ErrorList {
+	var allErrs field.ErrorList
+	eksaVersionPath := field.NewPath("spec").Child("EksaVersion")
+
+	if new.Spec.EksaVersion != nil && oldCluster.Spec.EksaVersion != nil {
+		parsedClusterVersion, err := semver.New(string(*oldCluster.Spec.EksaVersion))
+		if err != nil {
+			return allErrs
+		}
+
+		parsedUpgradeVersion, err := semver.New(string(*new.Spec.EksaVersion))
+		if err != nil {
+			allErrs = append(
+				allErrs,
+				field.Invalid(eksaVersionPath, new.Spec.EksaVersion, "upgrade cluster is invalid"))
+		}
+
+		majorVersionDifference := math.Abs(float64(parsedUpgradeVersion.Major) - float64(parsedClusterVersion.Major))
+		minorVersionDifference := float64(parsedUpgradeVersion.Minor) - float64(parsedClusterVersion.Minor)
+		supportedMinorVersionIncrement := float64(1)
+
+		// if major different or upgrade difference greater than one minor version
+		if majorVersionDifference > 0 || minorVersionDifference > supportedMinorVersionIncrement {
+			allErrs = append(
+				allErrs,
+				field.Invalid(eksaVersionPath, new.Spec.EksaVersion, fmt.Sprintf("cannot upgrade to %v from %v: EksaVersion upgrades must be sequential in minor versions", parsedUpgradeVersion, parsedClusterVersion)))
+		}
+
+		// allow "downgrades" if old version is greater than managment cluster
```

Comment:
If the mgmt cluster has v0.0.0 and you apply a workload with v0.1.0, then the change will get past the webhook, but it will fail at preClusterProviderReconcile because we can't have a workload with version greater than mgmt. 
If you try to correct the version (from 0.1.0 to 0.0.0), then webhook will block it because we don't want downgrades. 
But I think we should allow downgrades in this scenario where a successful upgrade never happened.